### DISCUSSION TITLE
Add a connect plugin

### DISCUSSION THREAD

**Discussion by Mehak-Mehta** [lowlighter/metrics#352] [Ideas]
Hi, I would like to request a new feature. Like a "Connect with me" plugin in which you can provide your socials (like twitter, Instagram, stackoverflow etc).
![2021-05-31 (2)](https://user-images.githubusercontent.com/76558546/120153127-f96c8680-c20b-11eb-9a2a-17f57e9f7517.png)

Something like this

**Comment by lowlighter** [comment]
It would be nice but unfortunately when links from SVGs are embed through `<img>` tags on GitHub, these are scrapped so you cannot actually click on them 😕 (that's why no plugins currently has links)

If this ever change I'm open to integrate it, but in current state you won't be able to do much

For now it's probably better to still rely on markdown links upon social buttons image

**Reply by jayvdb** [reply]
This should be possible now with the markdown output, and possibly the include in the 'insights' special output type.

Cross-linking https://github.com/lowlighter/metrics/discussions/547 which is about the hyperlinking problem.

**Reply by lowlighter** [reply]
Thinking about it, I'm not sure that a plugin dedicated to solely display links are that useful.

It should at least provide added value, as it seems less practical that directly adding links in the readme.md instead

**Reply by jayvdb** [reply]
What about allowing a github action variable that holds a mapping of unspecified contents, which could be used from the markdown template?

It could store the usernames/userids for social links, so users can share template snippets, similar to what I had in mind with https://github.com/lowlighter/metrics/discussions/630

**Reply by lowlighter** [reply]
Actually there's already something like this:
https://github.com/lowlighter/metrics/blob/4eeea4c0cd385c2c0a8d56c6b03b6484f8c61bdd/action.yml#L143-L148

But it's an old option which has no real use anymore in upstream templates.
It used to be required for `repository` template to pass the repo name (e.g. `query: '{"repo":"metrics"}'`) but now there is a top-level option `repo` which actually has the same effect. 

`query` (which is referred as `q` in the code) actually contains all users inputs (excepting tokens) including "inexistant options".

So basically I think we should just expose the `q` variable in resulting json `data` so it would allow to reuse any user input including custom ones

**Reply by jayvdb** [reply]
^ is now working on `@main`, so `query = {"x": 123}` in the yaml can be accessed as `q.x`, which is great for any unsupported usernames/ids.

`twitterUsername` and `user.login` and the other usernames via https://github.com/lowlighter/metrics/discussions/630 are all available.

Back to the OP, the second logo is dev.to and https://github.com/lowlighter/metrics/blob/master/source/plugins/posts/README.md has a dev.to username.

I worry about anyone wanting to link to Instagram from github, but meh .. maybe that is useful.  I dont recognise the 1st, 5th and 7th icon in the OP. 

Maybe the `posts` plugin could support a few sources, similar to the music plugin, as most people will have one active blog, or at least have one active blog that is most relevant to a Github audience.

**Reply by lowlighter** [reply]
> I worry about anyone wanting to link to Instagram from github, but meh .. maybe that is useful. I dont recognise the 1st, 5th and 7th icon in the OP.

I'm open to support it, but since I don't use instagram I won't implement it personally.
But if someone is motivated enough to create a new plugin for it, they're free to contribute 🙂 

> Maybe the posts plugin could support a few sources, similar to the music plugin, as most people will have one active blog, or at least have one active blog that is most relevant to a Github audience.

Same apply for this one. I know someone suggested to implement Mastodon (#318) but I don't know much about it.
Since personal blogs can take various forms, it's hard to generalize content extraction, I guess the best bet would be to use the RSS plugin for this one

**Comment by lowlighter** [comment]
Finally I think it's better to make users either put links directly in markdown or use the markdown template.
I don't think it brings a lot of value to be embed in an svg, especially since hyper-links cannot work in them atm
